src/index.ts

```diff
@@ -8,6 +8,7 @@ import { RoleArgs } from './descriptionToRoles'
 export interface Role<C, R, W> {
   matcher: (context: C, record: R, roleArgs: RoleArgs) => boolean
   queryConstraint: (context: C, roleArgs: RoleArgs) => W | boolean
+  requiredFields?: Record<string, any> | ((roleArgs: RoleArgs) => Record<string, any>)
```

Comment:
I feel like this is a bit in the no-mans-land when it comes to naming. Specific, but not specific enough.

Is it this?

```suggestion
  roleAuthorizationDependencies?: Record<string, any> | ((roleArgs: RoleArgs) => Record<string, any>)
```

Reply:
Or is it this?

```suggestion
  authorizationFields?: Record<string, any> | ((roleArgs: RoleArgs) => Record<string, any>)
```

Reply:
Maybe those are still too specific, but that's my lean there.

Reply:
Since this property is on authorisation rule, I felt like putting "authorization" into it was a bit redundant.
So essentially it's `prismaSelectObjectThatDescribesAuthorizationMatcherDependencies`. Now the question is how to make it a bit less Java-ish...
Maybe `matcherDependencies`? `matcherDependenciesSelect`?
@randytarampi @aertmann @cassiohub

Reply:
I'd prefer `select` too. It wouldn't scope it too much on `authorization` then.

Reply:
`matcherDependenciesSelect` could work or `matcherDependenciesFields` perhaps?